### ISSUE TITLE
Build Error on Debian

### ISSUE THREAD

**Issue by nabulator** [Mdashdotdashn/LittleGPTracker#4]
Using GCC 4.9.0 on a 64 bit machine.
First of all, I needed to add the -fpermissive flags since there's the "invalid conversion of a const pointer to a pointer".

```
Application/Instruments/SamplePool.cpp:8:
/home/nabulator/tools/LittleGPTracker/projects/../sources/Externals/Soundfont/DATATYPE.H:141:4: error: #error <datatype.h>: long is out of range!
   #error <datatype.h>: long is out of range!

```
It looks like the soundfont library disagrees about the fundamental datatype? I guess the library is designed for 32 bit compilers only?

Will attempt to force 32 bit compiling soon.

**Comment by Mdashdotdashn** [comment]
Yeah, the soundfont library dates from an age people didn't even know there might be 64-bit operating system some time.
I guess the easiest fix would be to disable soundfont if in a 64bit environment. The best one would be to 'retype' the library so it uses stdint's but that might be a hassle.

**Comment by ryanlaws** [comment]
I've been trying to get this to build for a few days, but I'm a total C++ noob so it's all been very naive. I went ahead and disabled soundfonts as well as updating some of the pointer types, added a couple null checks, etc. and now it doesn't segfault on song load, but I'm still not able to get any sound out of it w/ JACK (haven't tried other drivers).

**Comment by fgaz** [comment]
Isn't that a "false alarm" since that check was made for <32bit archs and whatever would fit in 32 bits trivially fits in 64 too? I've got a nixos package [here](https://github.com/NixOS/nixpkgs/pull/91766) that just disables the checks and it seems to work
